File: IgnoreReturnValues/IgnoreReturnValues/ViewController.m
//
//  ViewController.m
//  IgnoreReturnValues
//
//  Created by Ibrahim Hassan on 17/01/19.
//  Copyright © 2019 None. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    [self greet];
    NSLog(@"%@", [self greet]);
}

-(nullable NSString*) greet {
    return @"Hello";
}


@end

File: IgnoreReturnValues/IgnoreReturnValues-Swift/ViewController.swift
//
//  ViewController.swift
//  IgnoreReturnValues-Swift
//
//  Created by Ibrahim Hassan on 17/01/19.
//  Copyright © 2019 None. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        greet()
    }

    @discardableResult
    func greet() -> String? {
        return "Hello"
    }

}

